galsim/chromatic.py

```diff
@@ -1567,9 +1661,26 @@ def drawImage(self, bandpass, image=None, integrator='trapezoidal', **kwargs):
             jac, offset, _ = self._getTransformations(bandpass.red_limit)
             int_im = galsim.Transform(int_im, jac=jac, offset=offset, gsparams=self.gsparams)
             image = int_im.drawImage(image=image, **kwargs)
+            self._last_wcs = image.wcs
             return image
         else:
-            return ChromaticObject.drawImage(self, bandpass, image, integrator, **kwargs)
+            image = ChromaticObject.drawImage(self, bandpass, image, integrator, **kwargs)
+            self._last_wcs = image.wcs
+            return image
+
+    @property
+    def noise(self):
+        # Condition for being able to propagate noise:
+        # 1) All transformations are achromatic.
+        # 2) This ChromaticTransformation wraps a ChromaticConvolution with a valid noise property.
+        if (hasattr(self._jac, '__call__') or
+            hasattr(self._offset, '__call__') or
+            not self._flux_ratio._const):
+            raise TypeError("Cannot propagate noise through chromatic transformation")
+        noise = self.original.noise
+        jac = self._jac
+        flux_ratio = self._flux_ratio(42.) # const, so use any wavelength
+        return noise.transform(jac[0,0], jac[0,1], jac[1,0], jac[1,1])*flux_ratio**2
```

Comment:
Yes, although I'd forgotten that this, being a BaseCorrelatedNoise, would need the extra bits for the rng and wcs.  Up to you whether you want to keep the longer (but slightly more efficient) version or keep what you originally had.